docs/content/concepts/sui-move-concepts/conventions.mdx

```diff
@@ -0,0 +1,331 @@
+- `property_name`: Reads a property.
+- `destroy_empty`: Destroys an object or data structure that has values with the **drop** ability.
+- `to_object_name`: Transforms an Object X to Object Y.
+- `from_object_name`: Transforms an Object Y to Object X.
+- `property_name_mut`: Returns a mutable reference.
+- `property_name_immut`: Returns an immutable reference.
+
+## Potato Structs
+
+#### Do not call structs as Potato. It is a pattern recognized by the lack of abilities.
+
+```move
+module conventions::request {
+  // ✅ Right
+  struct Request {}
+
+  // ❌ Wrong
+  struct RequestPotato {}
+}
+```
+
+## Read Functions
+
+#### Be mindful of the dot syntax when naming functions. Avoid using the object name on function names.
+
+```move
+module conventions::lib {
+  struct Profile {
+    age: u64
+  }
+
+  // ✅ Right
+  public fun age(self: &Profile):  u64 {
+    self.age
+  }
+
+  // ❌ Wrong
+  public fun profile_age(self: &Profile): u64 {
+    self.age
+  }
+}
+
+module conventions::lib {
+  use conventions::lib::{Self, Profile};
+
+  public fun get_tokens(profile: &Profile) {
+
+  // ✅ Right
+  let name = profile.age();
+
+  // ❌ Wrong
+  let name2 = profile.profile_age();
+}
+```
+## Empty Function
+
+#### Functions that create data structures must be called empty.
+
+```move
+module conventions::data_structure {
+  struct DataStructure has copy, drop, store {
+    bits: vector<u8>
+  }
+
+  public fun empty(): DataStructure {}
+}
+```
+## New Function
+
+#### Functions that create objects must be called new.
+
+```move
+module conventions::object {
+  struct Object has key, store {
+    id: UID
+  }
+
+  public fun new(ctx:&mut TxContext): Object {}
+}
+```
+
+## Shared Objects
+
+#### Shared objects must be created via a new function and be shared in a separate function. The share function must be named share.
+
+```move
+module conventions::profile {
+  struct Profile has key {
+    id: UID
+  }
+
+  public fun new(ctx:&mut TxContext): Profile {}
+
+  public fun share(profile: Profile) {}
+}
+```
+
+## Reference Functions
+
+#### Functions that return a reference must be named property_name_mut or property_name_immut. The borrow is implied by the mut and immut.
+
+```move
+module conventions::profile {
+  struct Profile has key {
+    id: UID,
+    name: String,
+    age: u8
+  }
+
+  // profile.name_immut()
+  public fun name_immut(self: &Profile): &String {}
+
+  // profile.age_mut()
+  public fun age_mut(self: &mut Profile): &mut u8 {}
+}
+```
+
+## Separation of Concerns
+
+#### Modules must be designed around one Object or Data Structure. A variant structure should have its own module to avoid complexity and bugs.
+
+```move
+module conventions::wallet {
+  struct Wallet has key, store {
+    id: UID,
+    amount: u64
+  }
+}
+
+module conventions::claw_back_wallet {
+  struct Wallet has key {
+    id: UID,
+    amount: u64
+  }
+}
+```
+
+## Errors
+
+#### Errors must be CamelCase, start with an E and be descriptive.
```

Comment:
Just speculating and dreaming - I wish we adopted "ECamelCase" term 😆 
But I haven't seen anyone use it.

Reply:
Well, it will be a thing now :)